git-duet-prepare-commit-msg/main.go

```diff
@@ -45,21 +50,53 @@ func main() {
 		os.Exit(0)
 	}
 
-	var coAuthorsTrailer string
-	for _, c := range committers {
-		coAuthorsTrailer += "Co-authored-by: " + c.Name + " <" + c.Email + ">\n"
+	commitMsg, err := ioutil.ReadFile(commitMsgFile)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
 	}
-	coAuthorsTrailer = strings.TrimSuffix(coAuthorsTrailer, "\n")
 
-	commitMsg, err := ioutil.ReadFile(commitMsgFile)
+	coAuthorTrailerRegexp := regexp.MustCompile(`Co-authored-by:\s.+\s<.+>`)
+	trailerExists := coAuthorTrailerRegexp.Match(commitMsg)
+	if trailerExists && commitMsgSource != "commit" {
+		/* The goal here is to not add trailers in interactive rebasing or cherry-picking
+		   since authorship doesn't get changed. Since this hook doesn't know whether it is invoked
+		   as part of rebasing or cherry-picking, at the very least, it checks for existing trailers,
+		   and if there is one, no new trailers will be appended.
+		   Trailers will still be appended for "git commit --amend" in which case the
+		   commitMsgSource's value is "commit". */
+		os.Exit(0)
+	}
+
+	// override the default "addIfDifferentNeighbor" so that no duplicate trailers will get appended
+	err = gitConfig.SetUnnamespacedKey("trailer.ifexists", "addIfDifferent")
```

Comment:
I think setting this un-namespaced variable here might cause some slight confusion to people already depending on a different behavior. I admit it's probably pretty rare, but what do you think, rather than setting this here, we just document that you can set this value to avoid duplicate `Co-authored-by:`s for the same author?

Reply:
Right, makes sense. I removed this part from the hook and wrote it into the README so that the user can optionally set it.